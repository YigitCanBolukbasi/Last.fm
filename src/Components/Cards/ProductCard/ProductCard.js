import {View, Text, Image} from 'react-native';
import React from 'react';
import styles from './ProductCard.styles';
import {useContext} from 'react';
import ThemeContext from '../../../Contexts/ThemeContext';

const ProductCard = ({product}) => {
  const {theme} = useContext(ThemeContext);

  const img = product.image;
  let photos = '';
  img.forEach(i => {
    photos = i;
  });

  return (
    <View style={styles.shadow_container}>
      <View style={theme ? styles.container_dark : styles.container}>
        <Image
          style={styles.image}
          source={
            photos['#text']
              ? {
                  uri: photos['#text'],
                }
              : null
          }
        />
        <View style={styles.body_container}>
          <View style={styles.title_container}>
            <Text
              testID="product-title"
              style={theme ? styles.title_dark : styles.title}>
              Graduation
            </Text>
            <Text
              testID="product-name"
              style={theme ? styles.name_dark : styles.name}>
              {product.name}
            </Text>
          </View>
          <View style={styles.inner_container}>
            <Text style={theme ? styles.text_dark : styles.text}>
              {product.listeners && 'Listeners:'}
              {product.listeners && product.listeners}
            </Text>
            <Text
              testID="product-playcount"
              style={theme ? styles.text_dark : styles.text}>
              PlayCount:{product.playcount}
            </Text>
          </View>
        </View>
      </View>
    </View>
  );
};

export default ProductCard;
