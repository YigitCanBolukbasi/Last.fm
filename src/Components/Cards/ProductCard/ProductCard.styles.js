import {StyleSheet, Dimensions} from 'react-native';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 10,
    height: Dimensions.get('window').height / 9,
    zIndex: 1,
    backgroundColor: 'white',
  },
  shadow_container: {
    padding: 3,
    elevation: 10,
    marginBottom: 10,
    borderRadius: 12,
    shadowColor: '#000',
    margin: 5,
  },
  image: {
    width: 35,
    height: 35,
    borderRadius: 35,
    alignSelf: 'center',
    marginRight: 5,
  },
  body_container: {
    flex: 1,
    flexDirection: 'row',
    margin: 5,
  },
  title_container: {
    flex: 1,
    margin: 5,
  },
  inner_container: {
    flex: 1,
    margin: 5,
  },
  title: {
    fontSize: 10,
    fontWeight: '700',
    color: 'black',
    borderBottomColor: 'black',
    borderBottomWidth: 1,
  },
  name: {
    fontSize: 10,
    fontWeight: '700',
    color: 'black',
  },
  text: {
    fontSize: 8,
    color: 'black',
    margin: 5,
  },
});
