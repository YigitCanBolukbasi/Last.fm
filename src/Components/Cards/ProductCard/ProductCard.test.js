import React from 'react';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import ProductCard from './ProductCard';
import {ThemeProvider} from '../../../Contexts/ThemeContext';
const items = {
  '@attr': {
    rank: '20',
  },
  artist: {
    mbid: 'c8b03190-306c-4120-bb0b-6f2ebfc06ea9',
    name: 'The Weeknd',
    url: 'https://www.last.fm/music/The+Weeknd',
  },
  image: [
    {
      '#text':
        'https://lastfm.freetls.fastly.net/i/u/34s/2a96cbd8b46e442fc41c2b86b821562f.png',
      size: 'small',
    },
    {
      '#text':
        'https://lastfm.freetls.fastly.net/i/u/64s/2a96cbd8b46e442fc41c2b86b821562f.png',
      size: 'medium',
    },
    {
      '#text':
        'https://lastfm.freetls.fastly.net/i/u/174s/2a96cbd8b46e442fc41c2b86b821562f.png',
      size: 'large',
    },
    {
      '#text':
        'https://lastfm.freetls.fastly.net/i/u/300x300/2a96cbd8b46e442fc41c2b86b821562f.png',
      size: 'extralarge',
    },
  ],
  listeners: '341031',
  mbid: '05a472b0-09e5-44cd-bf2d-f6f6ab56cc32',
  name: 'What You Need',
  playcount: '2619123',
  streamable: '0',
  url: 'https://www.last.fm/music/The+Weeknd/_/What+You+Need',
};

test('should first', () => {
  const comp = render(
    <ThemeProvider>
      <ProductCard product={items} />
    </ThemeProvider>,
  );
  expect(comp).toMatchSnapshot();
});

test('should render title correctly', () => {
  const title = 'Graduation';
  const comp = render(
    <ThemeProvider>
      <ProductCard product={items} />
    </ThemeProvider>,
  );
  const artistTitle = comp.getByTestId('product-title').children[0];
  expect(artistTitle).toBe(title);
});
test('should render name correctly', () => {
  const name = items.name;
  const comp = render(
    <ThemeProvider>
      <ProductCard product={items} />
    </ThemeProvider>,
  );
  const artistName = comp.getByTestId('product-name').children[0];
  expect(artistName).toBe(name);
});

test('should render playcount correctly', () => {
  const playCount = 'PlayCount:';
  const comp = render(
    <ThemeProvider>
      <ProductCard product={items} />
    </ThemeProvider>,
  );
  const artistPlaycount = comp.getByTestId('product-playcount').children[0];
  expect(artistPlaycount).toBe(playCount);
});
