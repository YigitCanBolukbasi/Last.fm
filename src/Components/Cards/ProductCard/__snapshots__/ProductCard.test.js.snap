// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should first 1`] = `
<View
  style={
    Object {
      "borderRadius": 12,
      "elevation": 10,
      "margin": 5,
      "marginBottom": 10,
      "padding": 3,
      "shadowColor": "#000",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flexDirection": "row",
        "height": 148.22222222222223,
        "padding": 10,
        "zIndex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://lastfm.freetls.fastly.net/i/u/300x300/2a96cbd8b46e442fc41c2b86b821562f.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "borderRadius": 35,
          "height": 35,
          "marginRight": 5,
          "width": 35,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "margin": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "borderBottomColor": "black",
              "borderBottomWidth": 1,
              "color": "black",
              "fontSize": 10,
              "fontWeight": "700",
            }
          }
          testID="product-title"
        >
          Graduation
        </Text>
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 10,
              "fontWeight": "700",
            }
          }
          testID="product-name"
        >
          What You Need
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "margin": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 8,
              "margin": 5,
            }
          }
        >
          Listeners:
          341031
        </Text>
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 8,
              "margin": 5,
            }
          }
          testID="product-playcount"
        >
          PlayCount:
          2619123
        </Text>
      </View>
    </View>
  </View>
</View>
`;
